### ISSUE TITLE
Question: Signed JWT

### ISSUE THREAD

**Issue by alejandro-falkowski-gelato** [Nerzal/gocloak#163]
Hello,

Just wanted to raise a question and hopefully you can help. I am looking at Signed JWT feature of Keycloak as described here:

- https://www.keycloak.org/docs/latest/server_admin/#_client-credentials
- https://www.janua.fr/understanding-client-authenticator-security-with-keycloak/

Is this supported or in the pipeline?

**Comment by SVilgelm** [comment]
As far as I understand It's not supported yet, but if you can provide a curl command that get a token using a signed JWT, then I can help you.

**Comment by Nerzal** [comment]
right now it is not supported, but i'm willing to implement it. 
If you could help us with some curl examples or similar, it would greatly help :)

**Comment by alejandro-falkowski-gelato** [comment]
Thanks guys so after some research I have found the following information:

- https://developer.okta.com/docs/reference/api/oidc/#jwt-with-shared-key

So basically the following scenarios I would like to add 
- JWT with shared key
- JWT with private key

From what I can see we need to add some fields to token options:

```go
ClientAssertion     *string  `json:"client_assertion,omitempty"`
ClientAssertionType *string  `json:"client_assertion_type,omitempty"`
``` 

Then I am not really sure how to sign them using shared or private.

**Comment by alejandro-falkowski-gelato** [comment]
From OpenID connect point of view https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication

**Comment by SVilgelm** [comment]
Looks like I got an understanding how to use the signed JWT. I created a draft commit: #184
@alejandro-falkowski-gelato please look at it and share your thoughts.
I will play with different Login functions, but it would be great if you can give more details about your workflow, how are you going to use the Signed JWT

**Comment by SVilgelm** [comment]
Released in v6.2.0
